spec/requests/posts_controller_spec.rb

```diff
@@ -2049,6 +2049,76 @@
     end
   end
 
+  describe "#purge_revisions" do
+    before { SiteSetting.can_permanently_delete = true }
+
+    fab!(:post) do
+      Fabricate(
+        :post,
+        user: Fabricate(:user),
+        raw: "Lorem ipsum dolor sit amet, cu nam libris tractatos, ancillae senserit ius ex",
+      )
+    end
+
+    fab!(:post_with_no_revisions) do
+      Fabricate(
+        :post,
+        user: Fabricate(:user),
+        raw: "Lorem ipsum dolor sit amet, cu nam libris tractatos, ancillae senserit ius ex",
+      )
+    end
+
+    fab!(:post_revision) { Fabricate(:post_revision, post: post) }
+    fab!(:post_revision_2) { Fabricate(:post_revision, post: post) }
+
+    let(:post_id) { post.id }
+
+    describe "when logged in as a regular user" do
+      it "does not delete revisions" do
+        sign_in(user)
+        delete "/posts/#{post_id}/revisions/purge.json"
+        expect(response).to_not be_successful
+      end
+    end
+
+    describe "when logged in as staff" do
+      before { sign_in(admin) }
+
+      it "fails when post record is not found" do
+        delete "/posts/#{post_id + 1}/revisions/purge.json"
+        expect(response).to_not be_successful
+      end
+
+      it "fails when no post revisions are found" do
+        delete "/posts/#{post_with_no_revisions.id}/revisions/purge.json"
+        expect(response).to_not be_successful
+      end
+
+      it "fails when 'can_permanently_delete' setting is false" do
+        SiteSetting.can_permanently_delete = false
+        delete "/posts/#{post_id}/revisions/purge.json"
+        expect(response).to_not be_successful
+      end
+
+      it "permanently deletes revisions from post and adds a staff log" do
+        delete "/posts/#{post_id}/revisions/purge.json"
+        expect(response.status).to eq(200)
+
+        # It creates a staff log
+        logs =
+          UserHistory.create!(
```

Comment:
This reads odd to me... is it creating the log after the request?

Reply:
This was a mistake, updated.